### PR TITLE
Details basic auth VS BO

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | 4.5+, 3.3, ~~2.5~~
| Edition       | All

Explains a bit more why this recommendation is made, avoids misinterpretation about SiteAccess example and REST not using SiteAccess.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [x] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/api/rest_api/rest_api_authentication.md [hunk 1]
```diff
@@ -365,10 +365,13 @@ If the installation has a dedicated host for REST, you can enable HTTP basic aut
...
-    * If the Back Office SiteAccess matches localhost/admin, it will call the REST API under `//localhost/api/ibexa/v2`.
+    * If the Back Office SiteAccess matches `//admin.example.com` (through `Map\Host`, `HostElement` or `HostText`), it will call the REST API under `//admin.example.com/api/ibexa/v2`;
+    * If the Back Office SiteAccess matches `//localhost/admin` (through `URIElement`, `Map\URI` or `Regex\URI`), it will call the REST API under `//localhost/api/ibexa/v2` because SiteAccess matching with REST isn't enabled at URL level.
+
+    If you set up basic authentification for `pattern: ^/api/ibexa/v2` to have it on your front office on both production and developments environments, your development environment's Back Office doesn't work properly anymore.
+    This Back Office will try to access REST through the same URL than front office. Even if you're logged in your Back Office and the [X-SiteAccess header](rest_requests.md#siteaccess) is used, the firewall denies access to REST as you're not logged through basic authentification. Some
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_authentication.md:372]:
```suggestion
    This Back Office will try to access REST through the same URL than front office. Even if you're logged in your Back Office and the [X-SiteAccess header](rest_requests.md#siteaccess) is used, the firewall denies access to REST as you're not logged through basic authentification.
```

**Review by julitafalcondusza — APPROVED** [review]

File: docs/api/rest_api/rest_api_authentication.md [hunk 2]
```diff
@@ -365,10 +365,13 @@ If the installation has a dedicated host for REST, you can enable HTTP basic aut
 
     Back Office uses the REST API too (for some parts like the Location tree or the Calendar) on its own domain.
     
-    * If the Back Office SiteAccess matches `admin.example.com`, it will call the REST API under `//admin.example.com/api/ibexa/v2`;
-    * If the Back Office SiteAccess matches localhost/admin, it will call the REST API under `//localhost/api/ibexa/v2`.
+    * If the Back Office SiteAccess matches `//admin.example.com` (through `Map\Host`, `HostElement` or `HostText`), it will call the REST API under `//admin.example.com/api/ibexa/v2`;
+    * If the Back Office SiteAccess matches `//localhost/admin` (through `URIElement`, `Map\URI` or `Regex\URI`), it will call the REST API under `//localhost/api/ibexa/v2` because SiteAccess matching with REST isn't enabled at URL level.
+
+    If you set up basic authentification for `pattern: ^/api/ibexa/v2` to have it on your front office on both production and developments environments, your development environment's Back Office doesn't work properly anymore.
```
Comment (MagdalenaZuba) [inline docs/api/rest_api/rest_api_authentication.md:371]:
```suggestion
    If you enable basic authentication for pattern: ^/api/ibexa/v2 to use it in your front office across both production and development environments, your development environment's Back Office won't work correctly.
```

File: docs/api/rest_api/rest_api_authentication.md [hunk 3]
```diff
@@ -365,10 +365,13 @@ If the installation has a dedicated host for REST, you can enable HTTP basic aut
...
+    This Back Office will try to access REST through the same URL than front office. Even if you're logged in your Back Office and the [X-SiteAccess header](rest_requests.md#siteaccess) is used, the firewall denies access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.
```
Comment (MagdalenaZuba) [inline docs/api/rest_api/rest_api_authentication.md:372]:
```suggestion
    This Back Office tries to access REST through the same URL as the front office. Even when logged in Back Office  and using the [X-SiteAccess header](rest_requests.md#siteaccess), the firewall blocks access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.
```

**Review by MagdalenaZuba — APPROVED** [review]

File: docs/api/rest_api/rest_api_authentication.md [hunk 4]
```diff
@@ -365,10 +365,13 @@ If the installation has a dedicated host for REST, you can enable HTTP basic aut
...
-    * If the Back Office SiteAccess matches localhost/admin, it will call the REST API under `//localhost/api/ibexa/v2`.
+    * If the Back Office SiteAccess matches `//admin.example.com` (through `Map\Host`, `HostElement` or `HostText`), it will call the REST API under `//admin.example.com/api/ibexa/v2`;
+    * If the Back Office SiteAccess matches `//localhost/admin` (through `URIElement`, `Map\URI` or `Regex\URI`), it will call the REST API under `//localhost/api/ibexa/v2` because SiteAccess matching with REST isn't enabled at URL level.
+
+    If you enable basic authentication for pattern: ^/api/ibexa/v2 to use it in your front office across both production and development environments, your development environment's Back Office won't work correctly.
+    This Back Office tries to access REST through the same URL as the front office. Even when logged in Back Office  and using the [X-SiteAccess header](rest_requests.md#siteaccess), the firewall blocks access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_authentication.md:372]:
```suggestion
    This Back Office tries to access REST through the same URL as the front office. Even when logged in Back Office and using the [X-SiteAccess header](rest_requests.md#siteaccess), the firewall blocks access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.
```

**Comment by adriendupuis** [comment]
I'm not sure about 2.5. https://doc.ibexa.co/en/2.5/api/rest_api_guide/#configuration use the pattern that should be avoided. And the "you need to edit app/config/security.yml and uncomment the configuration block about REST" is disturbing: I do not found this block on 2.5.32.

**Pull request merged**